armory/utils/metrics.py

```diff
@@ -721,6 +721,47 @@ def video_tracking_mean_iou(y, y_pred):
     return mean_ious
 
 
+def video_tracking_mean_success_rate(y, y_pred):
+    """
+    Mean success rate averaged over all thresholds in {0, 0.05, 0.1, ..., 1.0} and all frames.
+    This function expects to receive a single video's labels/prediction as input.
+
+    y (List[Dict, ...]): list of length equal to number of examples. Each element
+                  is a dict with "boxes" key mapping to (num_frames, 4) numpy array
+    y_pred (List[Dict, ...]): list of length equal to number of examples. Each element
+                  is a dict with "boxes" key mapping to (num_frames, 4) numpy array
+    """
+    _check_video_tracking_input(y, y_pred)
+    if len(y_pred) > 1:
+        raise ValueError(f"y_pred expected to have length of 1, found {len(y_pred)}.")
+
+    thresholds = np.arange(0, 1.05, 0.05)
```

Comment:
Given that line 757 defines success to be iou > threshold, I don't think you should include a threshold of 1.  Success will always be False for that threshold which means the mean success rate will be artificially reduced.  On the other hand, if you change line 757 to be iou >= threshold, then maybe you shouldn't include a threshold of 0, because iou >= 0 is always True

Reply:
I had copied this from the other implementation but I agree with you, it didn't make sense to me either. I'll change it to stop at 0.95